ckanext/geocat/metadata.py

```diff
@@ -485,31 +519,18 @@ class GeocatDcatDownloadDistributionMetdata(GeocatDcatDistributionMetadata):
     def get_metadata(self, dist_xml, dataset_meta):
         dist = self._handle_single_distribution(dist_xml, dataset_meta)
 
-        # if a download URL ends with zip,
-        # assume the media type is application/zip, no matter what geocat says
-        try:
-            if dist['download_url'].endswith('.zip'):
-                dist['media_type'] = 'application/zip'
-        except (KeyError, AttributeError):
-            pass
-
         return dist
 
     def get_mapping(self):
         return {
             'name': XPathValue('.//gmd:name/gco:CharacterString/text()'),
             'protocol': XPathValue('.//gmd:protocol/gco:CharacterString/text()'),  # noqa
-            'language': StringValue(''),
-            'url': FirstInOrderValue(
-                [
-                    XPathValue('.//gmd:linkage//che:LocalisedURL[@locale = "#DE" and ./text()]/text()'),  # noqa
-                    XPathValue('.//gmd:linkage//che:LocalisedURL[@locale = "#FR" and ./text()]/text()'),  # noqa
-                    XPathValue('.//gmd:linkage//che:LocalisedURL[@locale = "#EN" and ./text()]/text()'),  # noqa
-                    XPathValue('.//gmd:linkage//che:LocalisedURL[@locale = "#IT" and ./text()]/text()'),  # noqa
-                    XPathValue('.//gmd:linkage//che:LocalisedURL[./text()]/text()'),  # noqa
-                    XPathValue('.//gmd:linkage//gmd:URL[./text()]/text()'),
-                ]
-            ),
+            'language': StringValue(''),  # noqa
+            # 'download_url' and 'url' are set to empty, their
+            # values will be determined and set later from 'url_list'
+            'download_url': StringValue(''),  # noqa
+            'url': StringValue(''),  # noqa
+            'url_list': XPathMultiValue('.//gmd:linkage//che:LocalisedURL[./text()]/text()'),  # noqa
```

Comment:
I think you must keep the FirstInOrderValue, as it's how we defined it: get the German URLs, if they don't exist take the French URLs etc. Otherwise you duplicate the distributions if they are supplied in different languages (which is probably always the case). But use `XPathMultiValue` instead of the `XPathValue` it was before.